client/core/helpers.go

```diff
@@ -432,105 +432,6 @@ func (ord *OrderReader) MatchReaders() []*matchReader {
 	return readers
 }
 
-// IsMaker returns if the user is the maker in this match.
```

Comment:
Looks to me like the rest of `matchReader` can go now. There are no invocations of `(*OrderReader).MatchReaders` remaining, and the remaining methods aren't used (some share names with `OrderReader` methods, but the `matchReader` variants aren't used.)